### ISSUE TITLE
More secure way to connect to Earth

### ISSUE THREAD

**Issue by rebroad** [zerotier/ZeroTierOne#1444]
Currently when I connect a MS Windows OS to ZeroTier Earth, I find that many of my ports are exposed (i.e. show as open or closed, rather than stealth, i.e. don't respond at all). It seems that the default firewall in Windows 10, 11, etc is not adequate to allow connections to ZeroTier Earth.

My quesiton is - how are people managing to connect to Earth in a safe way so that they cannot be port scanned, etc?

And is it possible to modify ZeroTier such that ZeroTier itself can provide this security - such as perhaps allowing the user to specify which ports they want to be exposed, rather than all of them (which is the current default)?

**Comment by joseph-henry** [comment]
> My quesiton is - how are people managing to connect to Earth in a safe way so that they cannot be port scanned, etc?

I'm not a regular Windows user but I'd suggest making sure your OS is up to date, that you turn off any servers you may be running and disable shared folders/volumes, etc. Maybe even make sure your hostname isn't betraying anything sensitive. Or maybe just join on a computer that isn't super sensitive.

> And is it possible to modify ZeroTier such that ZeroTier itself can provide this security - such as perhaps allowing the user to specify which ports they want to be exposed, rather than all of them (which is the current default)?

Not on Earth. If you ran your own network you could use our rules engine to segment the network and prevent certain types of connections but Earth is just a giant LAN party. Join at your own risk.

All that being said we aren't aware of any abuse on the network but I'd definitely take precautions.

**Comment by erikh** [comment]
if you join a zt network on windows 10, it'll come up with a huge network dialog that says, "do you want your pc to be discoverable?" press no.

try portscanning yourself after you do that. if you pressed yes, all hope abandon ye

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by rebroad** [comment]
> if you join a zt network on windows 10, it'll come up with a huge network dialog that says, "do you want your pc to be discoverable?" press no. try portscanning yourself after you do that. if you pressed yes, all hope abandon ye

Hi, The last time I tried this, and I made it a "public" network, which means it's supposed to be non-port-scannable, it ended up revealing a lot about my host when I port scanned it from another PC on the same network. Maybe I did something wrong, so I shall try this again.

@glimberg I was never claiming this was a bug - I raised this as a feature request, as I don't see why we can't make open ports something selective within the client.
